### PR TITLE
Make Shovel gracefully handle consumer_timeout exceptions

### PR DESCRIPTION
**Pull request by Haster2004**
## Proposed Changes

This PR is related to https://github.com/rabbitmq/rabbitmq-server/issues/4219 issue.
For some reason shovel consumer hit consumer_timeout. As a result channel is closed, but connection still present 
(your can see it on added picture: there is connection with zero chennels on it).
So in GUI it looks like shovel is OK (status is running), but it is impossible to send any messages by this shovel.

image

The problem is that shovel doesn't expect consumer_timeout, so it doesn't recreate connection correctly.
In this PR I have tried to fix this behaviour.

## Types of Changes

What types of changes does your code introduce to this project?
_Put an `x` in the boxes that apply_

- [x] Bug fix (non-breaking change which fixes issue #4219)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause an observable behavior change in existing systems)
- [ ] Documentation improvements (corrections, new content, etc)
- [ ] Cosmetic change (whitespace, formatting, etc)
- [ ] Build system and/or CI

## Checklist

_Put an `x` in the boxes that apply.
You can also fill these out after creating the PR.
If you're unsure about any of them, don't hesitate to ask on the mailing list.
We're here to help!
This is simply a reminder of what we are going to look for before merging your code._

- [x] I have read the `CONTRIBUTING.md` document
- [x] I have signed the CA (see https://cla.pivotal.io/sign/rabbitmq)
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] All tests pass locally with my changes
- [ ] If relevant, I have added necessary documentation to https://github.com/rabbitmq/rabbitmq-website
- [ ] If relevant, I have added this change to the first version(s) in release-notes that I expect to introduce it

## Further Comments

If this is a relatively large or complex change, kick off the discussion by explaining why you chose the solution you did and what alternatives you considered, etc.

### CODE REVIEW

File: deps/rabbitmq_shovel/src/rabbit_amqp091_shovel.erl [hunk 1]
```diff
@@ -230,6 +230,9 @@ handle_source({'EXIT', Conn, Reason},
               #{source := #{current := {Conn, _, _}}}) ->
     {stop, {inbound_conn_died, Reason}};
 
+handle_source({'EXIT', _Pid, {shutdown, {server_initiated_close, ?PRECONDITION_FAILED, Reason}}}, _State) ->
+    {stop, {inbound_consumer_no_ack, Reason}};
```
Comment (michaelklishin) [inline deps/rabbitmq_shovel/src/rabbit_amqp091_shovel.erl:234]: This return value of `inbound_consumer_no_ack` is not suitable for every `406 PRECONDITION FAILED` response. We should change it to `source_precondition_failed`

File: deps/rabbitmq_shovel/src/rabbit_amqp091_shovel.erl [hunk 2]
```diff
@@ -251,6 +254,9 @@ handle_dest(#'basic.cancel'{}, #{name := Name}) ->
 
 handle_dest({'EXIT', Conn, Reason}, #{dest := #{current := {Conn, _, _}}}) ->
     {stop, {outbound_conn_died, Reason}};
+    
+handle_dest({'EXIT', _Pid, {shutdown, {server_initiated_close, ?PRECONDITION_FAILED, Reason}}}, _State) ->
+    {stop, {outbound_consumer_no_ack, Reason}};
```
Comment (michaelklishin) [inline deps/rabbitmq_shovel/src/rabbit_amqp091_shovel.erl:259]: Just like above, it should be `destination_precondition_failed` or something generic like that.

**Review by michaelklishin — CHANGES_REQUESTED** [review]

**Comment by michaelklishin** [comment]
I will make the renaming part.

**Comment by Haster2004** [comment]
@michaelklishin , thanks a lot!

**Comment by michaelklishin** [comment]
This will ship in `3.10.7` and `3.9.22`.

**Pull request merged**